### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by glennj**
Previously, most of the exercises defined the test _data_, then dynamically created the actual tests. Too cute.

This pull request will do 3 things
1. move the boiler place to a separate file that gets sourced.
2. restructure the tests to "hardcode" them, which will be much simpler to follow
3. add "skips" like other tracks do (and document what to do about them).

Dear reviewer: I'm sorry. I've tried to make this as painless as possible. There will be one commit for each exercise, but there are 120 of them. I've commented to pinpoint some of the highlights. Rest assured that the new test scripts are in fact passing the checks.

### CODE REVIEW

**Comment by glennj** [comment]
The configlet lint check is currently failing: PR #206 addresses this.

File: exercises/practice/hello-world/hello-world.test [hunk 1]
```diff
@@ -1,40 +1,11 @@
...
-    }
-}
-
-proc cleanupTests {} {
-    set failed [failed]
-    uplevel 1 ::tcltest::cleanupTests
-    if {$failed} {exit 1}
-}
-
-if {$::argv0 eq [info script]} {
```
Comment (glennj) [inline exercises/practice/hello-world/hello-world.test:31]: The initial hello world test script clearly is simpler to read and less daunting.

File: exercises/practice/acronym/acronym.test [hunk 2]
```diff
@@ -1,52 +1,53 @@
 #!/usr/bin/env tclsh
 package require tcltest
 namespace import ::tcltest::*
+source testHelpers.tcl
+
+############################################################
 source "acronym.tcl"
 
-proc fail_fast {} {
```
Comment (glennj) [inline exercises/practice/acronym/acronym.test:6]: The acronym tests are a typical example: in the old version, there was a pile of boiler plate, the test data written as a "table" then the tests executed dynamically. Tests cannot be skipped, but the test script by default aborts at the first failure.

The new version is less clever, merely writing the tests out one-by-one, with a skip "annotation".

File: exercises/practice/acronym/testHelpers.tcl [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#############################################################
+# Override some tcltest procs with additional functionality
+
+# Allow an environment variable to override `skip`
+proc skip {patternList} {
```
Comment (glennj) [inline exercises/practice/acronym/testHelpers.tcl:5]: Each exercise now has this library of functions for the test script to use. 

The original `tcltest` module 
* does not have a mechanism to override the `skip` command.
* always returns a success exit status from `cleanupTests` regardless of how many tests failed.

File: exercises/practice/allergies/testHelpers.tcl [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#   expr {!!$a == !!$b}  ;# => 1
+#
+proc booleanMatch {expected actual} {
+    return [expr {
+        [string is boolean -strict $expected] &&
+        [string is boolean -strict $actual] &&
+        !!$expected == !!$actual
+    }]
+}
+customMatch boolean booleanMatch
```
Comment (glennj) [inline exercises/practice/allergies/testHelpers.tcl:44]: Here's an example of the testHelpers library providing a method for the tests to compare the expected/actual results more specifically than just string comparison.

File: bin/test_one_exercise [hunk 5]
```diff
@@ -32,6 +32,7 @@ echo "*** Test exercise $exercise:"
 echo
 
 cp "$solution" "$exercise.tcl"
-tclsh "$exercise.test" || exit 1
+
+RUN_ALL=yes tclsh "$exercise.test" || exit 1
```
Comment (glennj) [inline bin/test_one_exercise:36]: This automated testing script must specify this env var to override the "skips" in the test file.

File: exercises/practice/acronym/acronym.test (hunk 2 above)
Comment (ErikSchierboom) [reply exercises/practice/acronym/acronym.test:6]: I really like the new, simpler version. Makes it a lot easier to read and allowing for skips is great IMHO.

File: bin/test_one_exercise (hunk 5 above)
Comment (ErikSchierboom) [reply bin/test_one_exercise:36]: Nice. This does mean that the test runner also needs to be updated. It's an easy fix though, adding an `ENV` instruction that sets the above environment variable from within the [Dockerfile](https://github.com/exercism/tcl-test-runner/blob/main/Dockerfile). Want to submit a PR to the test runner once this is merged? If not, I'd be happy to do it for you.

File: exercises/practice/acronym/testHelpers.tcl (hunk 3 above)
Comment (ErikSchierboom) [reply exercises/practice/acronym/testHelpers.tcl:5]: Excellent! Less boilerplate for the student.

File: exercises/practice/allergies/testHelpers.tcl (hunk 4 above)
Comment (ErikSchierboom) [reply exercises/practice/allergies/testHelpers.tcl:44]: 👍

File: exercises/practice/hello-world/hello-world.test (hunk 1 above)
Comment (ErikSchierboom) [reply exercises/practice/hello-world/hello-world.test:31]: Great!

**Review by ErikSchierboom — APPROVED** [review]
Awesome work! I can't say I've meticulously studied all files, but I've checked some and read the comments.

**Pull request merged**